src/core/telemetry/utils.ts

```diff
@@ -5,11 +5,15 @@ import TelemetryReporter from "./telemetryReporter";
 import type { TelemetryEventMeasurements, TelemetryEventProperties } from "./telemetryReporterTypes";
 
 export function collectTelemetryEvent(event: string, properties?: TelemetryEventProperties, measurements?: TelemetryEventMeasurements) {
-  const reporter: TelemetryReporter = new TelemetryReporter(process.env.APPLICATIONINSIGHTS_CONNECTION_STRING);
-  reporter.sendTelemetryEvent(event, properties, measurements);
+  if (process.env.SWA_DISABLE_TELEMETRY == "false") {
+    const reporter: TelemetryReporter = new TelemetryReporter(aiKey);
+    reporter.sendTelemetryEvent(event, properties, measurements);
+  }
 }
 
 export function collectTelemetryException(exception: Error, properties?: TelemetryEventProperties, measurements?: TelemetryEventMeasurements) {
-  const reporter: TelemetryReporter = new TelemetryReporter(process.env.APPLICATIONINSIGHTS_CONNECTION_STRING);
-  reporter.sendTelemetryException(exception, properties, measurements);
+  if (process.env.SWA_DISABLE_TELEMETRY == "false") {
```

Comment:
Q: will this environment variable be set/unset if we are setting the telemetry flag through the swa command.

Reply:
yes